packages/phone-number-privacy/combiner/src/server.ts

```diff
@@ -87,3 +88,41 @@ export function createHandler<R extends OdisRequest>(
 ): PromiseHandler<R> {
   return meteringHandler(catchErrorHandler(enabled ? handler : disabledHandler<R>))
 }
+
+export function startProxy(req: any, res: any, config: CombinerConfig) {
+  const proxy = httpProxy.createProxyServer({})
+  let destinationUrl
+  const originalPath = req.path
+  const strippedPath = originalPath.replace(/\/combiner/, '')
+
+  switch (config.deploymentEnv) {
+    case 'mainnet':
+      // XXX (soloseng):URL may need to be updated after gen2 function is created on mainnet
+      destinationUrl =
+        'https://us-central1-celo-pgpnp-mainnet.cloudfunctions.net/combinerGen2' +
+        proxy.web(req, res, { target: destinationUrl })
+      break
+    case 'alfajores':
+      // XXX (soloseng):URL may need to be updated after gen2 function is created on alfajores
+      destinationUrl =
+        'https://us-central1-celo-phone-number-privacy.cloudfunctions.net/combinerGen2' +
+        strippedPath
+
+      proxy.web(req, res, { target: destinationUrl })
+      break
+    case 'staging':
+      destinationUrl =
+        'https://us-central1-celo-phone-number-privacy-stg.cloudfunctions.net/combinerGen2' +
+        strippedPath
+
+      proxy.web(req, res, { target: destinationUrl })
+      break
+  }
+  proxy.on('error', (_) => {
+    res
+      .status(500)
+      .send(
+        'Error in Proxying request to Combiner. Please make sure you are running the latest SDK version?'
```

Comment:
mmmm technically the user won't know that this is a proxy, and that could be misleading.

I would log that error for our logs, but here maybe it's better to return the same error that we had in a 500 of the v1.

Reply:
I added the internal logging.

I feel that it would be usefull to provide an error that provides the user with usefull feedback, hence the notice to update to the latest SDK.

The alternative would be `ErrorMessage.UNKNOWN_ERROR`, and I feel it's a bit obscure.